packages/a11y/src/useAccessibleForm.ts

```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react';
+import { IdAllocator } from '@leafygreen-ui/lib';
+
+const inputIdAllocator = IdAllocator.create('input');
+const labelIdAllocator = IdAllocator.create('label');
+
+/**
+ * Hook that returns two sets of props that accessibly associate a label and its respective input element
+ * @param id Describes the input element
+ */
+const useAccessibleForm = (idProp: string) => {
+  const inputId = useMemo(() => idProp ?? inputIdAllocator.generate(), [
```

Comment:
On the off chance someone changes the idProp, do we want to update this value? I'd imagine so.

Reply:
Yes we would want to update this value, which is the current behavior here. Just added tests to confirm.